### ISSUE TITLE
Dialog

### ISSUE THREAD

**Issue by lakotka-yura** [primefaces/primereact#1886]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
[ ] feature request
[ ] support request

**Current behavior**
Component <Dialog>. Open a modal dialog, open another modal dialog from it, then close it and open it again. The second time a new dialog opens behind the first dialog. In PrimeReact v6.1.0 - works correctly.

**Expected behavior**
Component <Dialog>. Open a modal dialog, open another modal dialog from it, then close it and open it again. A new dialog should open in front of the first dialog for the second and subsequent times.

**Minimal reproduction of the problem with instructions**
Open a modal dialog, open another modal dialog from it, then close it and open it again.
**Please tell us about your environment:**
Windows10, WebStorm2020.3.3, npm v7.6.3, node v15.12.0

* **React version:**
17.0.1

* **PrimeReact version:**
6.2.0, 6.2.1

* **Browser:** Chrome 89.0.4389.90 (64 бит) 
 
* **Language:** [all]

**Comment by mertsincan** [comment]
Hi,

Unfortunately, I couldn't replicate this issue. Could you please attach a codesandbox link for us to replicate?
My test; https://codesandbox.io/s/gallant-firefly-ych4p?file=/src/demo/DialogDemo.js

Best Regards,

**Comment by lakotka-yura** [comment]
Hi,
Unfortunately, I couldn't replicate this issue into a codesandbox too.
https://codesandbox.io/s/zealous-jang-198fn?file=/src/index.js.

In my project, I noticed 2 behavioral features:

1) The first dialog opens from the context menu of the tree. The context menu has z-index = 1001. The first dialog has z-index = 1002. The dialog contains a table. By double-clicking on a line, I open a second dialog. It has z-index = 1002. I close it. By double-clicking on the line, I open the second dialog again. It has z-index = 1001. The first dialog has z-index = 1002.

2) The first dialog opens from the context menu of the tree. The context menu has z-index = 1001. The first dialog has z-index = 1002. The dialog contains a table. There is a button in the footer of the table. By clicking on the button, I open a second dialog. It has z-index = 1003. I close it. By clicking on the button, I open the second dialog again. It has z-index = 1002. The first dialog has z-index = 1002.

When using PrimeReact v6.0.0, the context menu has a different z-index with each subsequent call. The z-index of the rest of the dialogs is set correctly.

**Comment by lakotka-yura** [comment]
Hi

I was able to reproduce the error.

https://codesandbox.io/s/zealous-jang-198fn?file=/src/index.js

In this code, this error occurs when adding content to the Dialog in the SecondDialog. It works correctly when deleting content.

**Comment by mertsincan** [comment]
Hi @lakotka-yura,
Thanks a lot! I'll work on it and get back to you.

**Comment by mertsincan** [comment]
Fixed in #1924.

Best Regards,
